lms/djangoapps/course_home_api/outline/views.py

```diff
@@ -268,8 +268,11 @@ def get(self, request, *args, **kwargs):  # pylint: disable=too-many-statements
                     'Please contact your degree administrator or '
                     '{platform_name} Support if you have questions.'
                 ).format(platform_name=settings.PLATFORM_NAME)
-            elif course_is_invitation_only(course):
+            elif CourseEnrollment.is_enrollment_closed(request.user, course_overview):
                 enroll_alert['can_enroll'] = False
+            elif CourseEnrollment.objects.is_course_full(course_overview):
+                enroll_alert['can_enroll'] = False
+                enroll_alert['extra_text'] = _('Course is full')
```

Comment:
I noticed the legacy course about page had logic for this (and 36 courses in prod - including some recent ones - have a max-enrollment setting that would trip this). So seemed sensible to include here too. This does do a database query.

Reply:
(Also, that string is what the old course about page showed in this case. I just stole it.)